repos/system_upgrade/el7toel8/actors/peseventsscanner/libraries/library.py

```diff
@@ -324,46 +350,54 @@ def add_output_pkgs_to_transaction_conf(transaction_configuration, events):
     api.current_logger().debug(message)
 
 
-def filter_out_transaction_conf_pkgs(to_install, to_remove, transaction_configuration):
+def filter_out_transaction_conf_pkgs(tasks, transaction_configuration):
     """
-    Filter out those PES events that conflict with the higher priority transaction configuration files.
+    Filter out those PES events conflicting with the higher priority transaction configuration files.
 
-    :param to_install: A dict {<pkg_name>: <repository>} with pkgs to install based on PES data
-    :param to_remove: A dict {<pkg_name>: <repository>} with pkgs to remove based on PES data
+    :param tasks: A dict with three dicts holding pkgs to keep, to install and to remove
     :param transaction_configuration: RpmTransactionTasks model instance with pkgs to install, keep and remove based
                                       on the user configuration files
     """
-    # The two lists below are for debug message purposes only
-    pes_based_pkgs_not_to_be_installed = []
-    pes_based_pkgs_not_to_be_removed = []
-
-    for pes_based_pkg_to_install in to_install.keys():
-        if pes_based_pkg_to_install in transaction_configuration.to_remove:
-            to_install.pop(pes_based_pkg_to_install)
-            pes_based_pkgs_not_to_be_installed.append(pes_based_pkg_to_install)
-    for pes_based_pkg_to_remove in to_remove.keys():
-        if pes_based_pkg_to_remove in transaction_configuration.to_install + transaction_configuration.to_keep:
-            to_remove.pop(pes_based_pkg_to_remove)
-            pes_based_pkgs_not_to_be_removed.append(pes_based_pkg_to_remove)
-
-    if pes_based_pkgs_not_to_be_installed:
+    pkgs_not_to_be_kept = []
+    pkgs_not_to_be_installed = []
+    pkgs_not_to_be_removed = []
+
+    for pkg_to_keep in tasks['to_keep'].keys():
```

Comment:
the `keys()` method is not needed here.
```suggestion
    for pkg_to_keep in tasks['to_keep']:
```

Same works in cases below.